src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -115,19 +117,32 @@ public static IEnumerable<Method> BuildJsonSerializationMethods(JsonObjectSerial
                 }
             }
 
+            if (interfaces is null && Configuration.UseModelReaderWriter)
```

Comment:
I saw you defined an option `use-write-core` to indicate if we will use `write-core-strategy`, It seems we need to use this option to identify if we need to create core method.

Reply:
Yes, the flag is to control whether we have writecore. But the condition here is a special handling for ResourcedData and Resource.